bluesky/abc.py

```diff
@@ -0,0 +1,264 @@
+try:
+    from typing import Protocol, runtime_checkable
+except ImportError:
+    from typing_extensions import Protocol, runtime_checkable
+
+from typing import Dict, Any, Optional, Callable, Generator, List, Union, Tuple, NoReturn
+
+
+Configuration = Dict[str, Dict[str, Any]]
+
+
+@runtime_checkable
+class Status(Protocol):
```

Comment:
I think there is also `watch(watcher)`: https://nsls-ii.github.io/ophyd/status.html#ophyd.status.DeviceStatus.watch
At least I needed it here: https://github.com/thomascobb/bluefly/blob/master/bluefly/core.py#L122

Reply:
this is considered optional:

per @danielballan:

> In this case the docs are just wrong. The watch method is optional. If not implemented, there will be no progress bar, but it will otherwise operate fine.
https://github.com/bluesky/bluesky/blob/566dce9e7ada95e2346c9aa61206c6a2f0a189bf/bluesky/utils/__init__.py#L1221

We could make a `Watchable` abc if we want (also the signature of the watcher function is a bit chaotic to represent because there are a lot of arguments that need to be accepted, but not all given)